examples/flax/image-captioning/run_image_captioning_flax.py

```diff
@@ -424,14 +424,12 @@ def main():
 
     # Handle the repository creation
     if training_args.push_to_hub:
-        if training_args.hub_model_id is None:
-            repo_name = get_full_repo_name(
-                Path(training_args.output_dir).absolute().name, token=training_args.hub_token
-            )
-        else:
-            repo_name = training_args.hub_model_id
-        create_repo(repo_name, exist_ok=True, token=training_args.hub_token)
-        repo = Repository(training_args.output_dir, clone_from=repo_name, token=training_args.hub_token)
+        # Retrieve of infer repo_name
+        repo_name = training_args.hub_model_id or Path(training_args.output_dir).absolute().name
```

Comment:
We don't use the `or` syntax in Transformers and always use explicit tests for `is None` as we do not want to rely on Python bool conversion magic :-) This applies to all other examples as well.

Reply:
Done in https://github.com/huggingface/transformers/pull/25120/commits/dd04c760a08aa6ef60bfb28ca3c1109912c08a2a

I've updated the examples to be explicit like this:
```py
       repo_name = training_args.hub_model_id
       if repo_name is None:
           repo_name = Path(training_args.output_dir).absolute().name
```